### PR TITLE
Mark methods and fields needed for types with debugger attributes.

### PR DESCRIPTION
**Pull request by erozenfeld**
1. Keep all fields and methods in types with DebuggerDisplayAttribute.
2. Keep all fields and methods in proxy types from DebuggerTypeProxyAttribute.

### CODE REVIEW

**Comment by erozenfeld** [comment]
Details about these attributes are here: https://msdn.microsoft.com/en-us/library/x810d419(v=vs.110).aspx
These attributes can also be specified at assembly level, I will address that in a separate PR.

**Comment by erozenfeld** [comment]
@marek-safar PTAL
/cc: @ericstj, @sbomer

**Comment by joncham** [comment]
Do we want to always preserve all of these? Should we have a switch to indicate if a debug/release build is occurring to allow for better debugability vs smallest size?

**Comment by erozenfeld** [comment]
@joncham Yes, that makes sense. I added a -v switch to control this.

File: linker/Mono.Linker.Steps/MarkStep.cs [hunk 1]
```diff
@@ -623,6 +629,27 @@ void MarkXmlSchemaProvider (TypeDefinition type, CustomAttribute attribute)
 			MarkNamedMethod (type, method_name);
 		}
 
+		void MarkTypeWithDebuggerDisplayAttribute (TypeDefinition type, CustomAttribute attribute)
+		{
+			if (_context.KeepMembersForDebuggerAttributes) {
+				MarkMethods (type);
```
Comment (marek-safar) [inline linker/Mono.Linker.Steps/MarkStep.cs]: I think this is just too aggressive it'll persist all methods/fields even if they are unused. You could try to evaluate content of {  } and only fallback to all if it's not simple property/field/method call

Comment (erozenfeld) [inline linker/Mono.Linker.Steps/MarkStep.cs]: @marek-safar I implemented what you suggested. PTAL

**Pull request merged**